outlines/models/openai.py

```diff
@@ -158,7 +182,7 @@ def __call__(
                 )
             )
         if "gpt-" in self.config.model:
-            return generate_chat(prompt, self.client, config)
+            return self.generate_chat(prompt, config)
```

Comment:
What I would do instead is move the response + token count extraction here instead of creating a new method.